File: src/SOP8/광양/EgovLogin/base-utility/src/main/java/egovframework/base/response/ResponseManager.java
package egovframework.base.response;

import java.util.List;
import egovframework.base.response.MessageResultListData;

public class ResponseManager {

    public static <T> MessageResultListData<T> makeResultList(List<T> datas, String errorMessage) {
        if (datas == null) {
            return new MessageResultListData<>(false, errorMessage);
        }

        MessageResultListData<T> response = new MessageResultListData<>(true, "");
        response.setDatas(datas);
        return response;
    }

    public static <T> MessageResultData<T> makeResult(T data, String errorMessage) {
        if (data == null) {
            return new MessageResultData<>(false, errorMessage);
        }

        MessageResultData<T> response = new MessageResultData<>(true, "");
        response.setData(data);
        return response;
    }
}

File: src/SOP8/Utility/Response/ResponseManager.cs
using System.Collections.Generic;

namespace Response
{
    public static class ResponseManager
    {
        public static MessageResultListData<T> MakeResultList<T>(IEnumerable<T> datas, string strErrorMessage)
        {
            if (datas == null)
                return new MessageResultListData<T>(false, strErrorMessage);

            MessageResultListData<T> response = new MessageResultListData<T>(true, "");
            response.Datas = datas;
            return response;
        }

        public static MessageResultData<T> MakeResult<T>(T data, string strErrorMessage)
        {
            if (data == null)
                return new MessageResultData<T>(false, strErrorMessage);

            MessageResultData<T> response = new MessageResultData<T>(true, "");
            response.Data = data;
            return response;
        }
    }
}
